fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java

```diff
@@ -0,0 +1,502 @@
+            throw new DdlException("Unknown metastore table type.");
+        }
+        switch (metastoreTableType) {
+            case "VIRTUAL_VIEW":
+                throw new DdlException("VIRTUAL_VIEW table is not supported.");
+            case "EXTERNAL_TABLE":
+            case "MANAGED_TABLE":
+                break;
+            default:
+                throw new DdlException("unsupported hudi table type [" + metastoreTableType + "].");
+        }
+
+        this.resourceName = resourceName;
+        copiedProps.remove(HUDI_RESOURCE);
+
+        String hudiBasePath = metastoreTable.getSd().getLocation();
+        if (!Strings.isNullOrEmpty(hudiBasePath)) {
+            hudiProperties.put(HUDI_BASE_PATH, hudiBasePath);
+        }
+
+        String hudiTableType = metastoreTable.getParameters().get("type");
+        if (!Strings.isNullOrEmpty(hudiTableType)) {
+            hudiProperties.put(HUDI_TABLE_TYPE, hudiTableType);
+        }
+
+        String hudiTablePrimaryKey = metastoreTable.getParameters().get("primaryKey");
+        if (!Strings.isNullOrEmpty(hudiTablePrimaryKey)) {
+            hudiProperties.put(HUDI_TABLE_PRIMARY_KEY, hudiTablePrimaryKey);
+        }
+
+        String hudiTablePreCombineField = metastoreTable.getParameters().get("preCombineField");
+        if (!Strings.isNullOrEmpty(hudiTablePreCombineField)) {
+            hudiProperties.put(HUDI_TABLE_PRE_COMBINE_FIELD, hudiTablePreCombineField);
+        }
+
+        String hudiInputFormat = metastoreTable.getSd().getInputFormat();
+        if (!Strings.isNullOrEmpty(hudiInputFormat)) {
+            hudiProperties.put(HUDI_TABLE_INPUT_FORMAT, hudiInputFormat);
+        }
+
+        List<FieldSchema> unPartHiveColumns = metastoreTable.getSd().getCols();
+        List<FieldSchema> partHiveColumns = metastoreTable.getPartitionKeys();
+        for (FieldSchema partHiveColumn : partHiveColumns) {
+            String columnName = partHiveColumn.getName();
+            Column partColumn = this.nameToColumn.get(columnName);
+            if (partColumn == null) {
+                throw new DdlException("partition column [" + columnName + "] must exist in column list");
+            } else {
+                this.partColumnNames.add(columnName);
+            }
+        }
+
+        for (FieldSchema s : unPartHiveColumns) {
+            this.dataColumnNames.add(s.getName());
+        }
+
+        Configuration conf = new Configuration();
+        HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(conf).setBasePath(hudiBasePath).build();
+        TableSchemaResolver schemaUtil = new TableSchemaResolver(metaClient);
+        Schema tableSchema;
+        try {
+            tableSchema = HoodieAvroUtils.createHoodieWriteSchema(schemaUtil.getTableAvroSchemaWithoutMetadataFields());
+        } catch (Exception e) {
+            throw new DdlException("Cannot get hudi table schema.");
+        }
+        for (Column column : this.fullSchema) {
```

Comment:
check column is allowed null or not like hive table ,since parquet do not support non-nullable column now

Reply:
done

Reply:
> return error if null or empty during create table

this is fetched from metastore info and it is non-null property